### ISSUE TITLE
ArmCord just loads endlessly it does not load into discord.

### ISSUE THREAD

**Issue by Jaoheah** [smartfrigde/armcord#513]
**Describe the bug**

ArmCord is not loading into Discord, it just loops forever. 

**To Reproduce**

1. Launch Armcord

For me its easy to reproduce it, as it won't load no matter how many times I open it, I am not sure if someone will be able to reproduce it outside of my system.

**Expected behavior**

I expected ArmCord to load into Discord, meaning that all of the Discord servers I am in will be on the left side of the screen, and so on, and so forth.

**Screenshots**

Here's a video: 

https://www.youtube.com/watch?v=ZJLdLY-HaRA

Me running it normally has the same effect as running it in a terminal.

**Desktop (please complete the following information):** 

```
OS: Garuda Linux x86_64
Host: OMEN by HP Laptop 17-ck1xxx
Kernel: 6.6.1-zen1-1-zen
Uptime: 2 hours, 33 mins
Battery: 100% [Full]

Packages: 2356 (pacman)[stable]
Shell: fish 3.6.1
Terminal: konsole 23.8.3

CPU: 12th Gen Intel(R) Core(TM) i7-12700H (20) @ 4.6z
GPU: Intel Iris Xe Graphics
GPU: NVIDIA GeForce RTX 3060 Mobile / Max-Q
Memory: 7.48 GiB / 31.02 GiB (24%)
```

 - Method of installation: chaotic-aur
 - Version: chaotic-aur/armcord-git r1003.641af3b-1 [0B 7.68MiB] [Installed] 

**Additional context**

Someone had a similar issue but closed the issue post https://github.com/ArmCord/ArmCord/issues/508

The three same things that me and Hues-Sueh have in common is that we both use Wayland, we both have NVIDIA GPUs, and we both are using the Vencord mod. 

This is what the terminal spat out when I ran `armcord` in the terminal

```
Wayland detected, using PipeWire for video capture.
Config is fine
[Config manager] doneSetup: true
[Config manager] performanceMode: none
ArmCord has been run before. Skipping setup.
No performance modes set
[Config manager] windowStyle: default
[Config manager] armcordCSP: true
[Config manager] doneSetup: true
[Config manager] customIcon: /usr/share/armcord/app.asar/assets/desktop.png
Setting up CSP unstricter...
[Config manager] trayIcon: default
[Config manager] customIcon: /usr/share/armcord/app.asar/assets/desktop.png
[Config manager] tray: false
[Config manager] skipSplash: false
[Config manager] tray: false
[Config manager] windowStyle: default
[Config manager] mods: vencord
[Config manager] noBundleUpdates: undefined
Downloading mod bundle
[Config manager] mods: vencord
https://github.com/Vendicated/Vencord/releases/download/devbuild/browser.js
[Mod loader] Loaded ArmCord Mod Loader made by Vendicated
(node:522604) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected end of JSON input
at JSON.parse (<anonymous>)
at getWindowState (/usr/share/armcord/app.asar/ts-out/utils.js:255:27)
at createCustomWindow (/usr/share/armcord/app.asar/ts-out/window.js:308:50)
at init (/usr/share/armcord/app.asar/ts-out/main.js:92:53)
at async /usr/share/armcord/app.asar/ts-out/main.js:107:9
(Use `electron --trace-warnings ...` to show where the warning was created)
(node:522604) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
WebContents #1 called ipcRenderer.sendSync() with 'modInstallState' channel without listeners.
WebContents #1 called ipcRenderer.sendSync() with 'get-app-version' channel without listeners.
[522642:1117/120609.764278:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
Error occurred in handler for 'getLang': Error: No handler registered for 'getLang'
at WebContents.<anonymous> (node:electron/js2c/browser_init:2:89250)
at WebContents.emit (node:events:513:28)
[522642:1117/120635.795901:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[522642:1117/120635.797054:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
```

**Comment by rojidev** [comment]
i recently had the same or a similar issue where armcord would load infinitely with the discord logo screen.
deleting the `window.json` file in the storage folder in armcord's config (`~/.config/ArmCord/storage` for me) fixed it for me ^ ^

the difference between a working `window.json` and the broken one seems like the broken one was simply missing brackets `{}` lol. you can recreate the issue by simply making `window.json` be an empty file

**Comment by Jaoheah** [comment]
> i recently had the same or a similar issue where armcord would load infinitely with the discord logo screen.
> deleting the `window.json` file in the storage folder in armcord's config (`~/.config/ArmCord/storage` for me) fixed it for me ^ ^
> 
> the difference between a working `window.json` and the broken one seems like the broken one was simply missing brackets `{}` lol

Ah, I unfortunately can't test this as I have since deleted it and instead installed the flatpak version.

**Comment by AlephNought0** [comment]
> i recently had the same or a similar issue where armcord would load infinitely with the discord logo screen. deleting the `window.json` file in the storage folder in armcord's config (`~/.config/ArmCord/storage` for me) fixed it for me ^ ^
> 
> the difference between a working `window.json` and the broken one seems like the broken one was simply missing brackets `{}` lol. you can recreate the issue by simply making `window.json` be an empty file

This does indeed fix this issue for me. Since the solution seems to exist I believe this issue should be closed.

**Comment by starchyunderscore** [comment]
> i recently had the same or a similar issue where armcord would load infinitely with the discord logo screen. deleting the `window.json` file in the storage folder in armcord's config (`~/.config/ArmCord/storage` for me) fixed it for me ^ ^
> 
> the difference between a working `window.json` and the broken one seems like the broken one was simply missing brackets `{}` lol. you can recreate the issue by simply making `window.json` be an empty file

I had the same issue on arch, with both the armcord-bin and armcord-git packages. This fix worked.

**Comment by Jaoheah** [comment]
It happened again on the Flatpak version. but I needed to uninstall it as it doesn't seem to create a config in ~/.config/ArmCord, so It can't be fixed via the method above this comment.

**Comment by Jaoheah** [comment]
>  Since the solution seems to exist I believe this issue should be closed.

It should be closed when whatever is causing the issue is fixed by the developers so it doesn't happen in the first place imo.

**Comment by AlephNought0** [comment]
> >  Since the solution seems to exist I believe this issue should be closed.
> 
> It should be closed when whatever is causing the issue is fixed by the developers so it doesn't happen in the first place imo.
> 

You're actually right. I have assumed that the user was tinkering with the config and that this might be the cause of the problem. I was tinkering with my config a lot too and when I saw this post I assumed I and the user were the problem and didn't think about the possibility of the application being the problem. My bad.

**Comment by Heus-Sueh** [comment]
> i recently had the same or a similar issue where armcord would load infinitely with the discord logo screen. deleting the `window.json` file in the storage folder in armcord's config (`~/.config/ArmCord/storage` for me) fixed it for me ^ ^
> 
> the difference between a working `window.json` and the broken one seems like the broken one was simply missing brackets `{}` lol. you can recreate the issue by simply making `window.json` be an empty file

directory in flatpak version:
`/var/home/heus/.var/app/xyz.armcord.ArmCord/config/ArmCord/storage`

**Comment by Jaoheah** [comment]
If you are encountering a similar issue to this but it just loads into a gray screen instead of looping endlessly check out this issue here: https://github.com/ArmCord/ArmCord/issues/540

**Comment by smartfrigde** [comment]
Fixed in 3.2.6
